src/server/test/routes/compareReadingsParamsTest.js

```diff
@@ -0,0 +1,321 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const { expect } = require('chai');
+const { chai, mocha, app, testDB } = require('../common');
+const { validateString, testInvalidField } = require('../util/validationHelpers');
+
+mocha.describe('Compare Readings Parameter Validation', () => {
+
+    mocha.describe('GET /api/compareReadings/meters/:meter_ids - Meter Compare Readings', () => {
+        const BASE_METER_ENDPOINT = '/api/compareReadings/meters';
+        
+        const validQuery = {
+            curr_start: '2023-01-01T00:00:00.000Z',
+            curr_end: '2023-01-02T00:00:00.000Z',
+            shift: 'P1D',
+            graphicUnitId: '1'
+        };
+
+        mocha.describe('URL Parameter Validation - meter_ids', () => {
+            mocha.it('should accept valid single meter ID', async () => {
+                const res = await chai.request(app)
+                    .get(`${BASE_METER_ENDPOINT}/1`)
+                    .query(validQuery);
+                
+                // Should return 200 or 500 (DB error) - no auth required for reading
+                expect([200, 500]).to.include(res.status);
+            });
+
+            mocha.it('should accept valid comma-separated meter IDs', async () => {
+                const res = await chai.request(app)
+                    .get(`${BASE_METER_ENDPOINT}/1,2,3`)
+                    .query(validQuery);
+                
+                expect([200, 500]).to.include(res.status);
+            });
+
+            mocha.it('should reject invalid meter ID patterns', async () => {
+                const invalidPatterns = [
```

Comment:
Does this apply across several routes so should be centralized? Also, could the map pattern be the base for this pattern with the added ones?